src/extensions/windows/Microsoft.DotNet.UpgradeAssistant.Extensions.Windows/UWPtoWinAppSDKUpgrade/WinUIBackButtonAnalyzer.cs

```diff
@@ -0,0 +1,76 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace Microsoft.DotNet.UpgradeAssistant.Extensions.Windows
+{
+    [ApplicableComponents(ProjectComponents.WinUI)]
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public sealed class WinUIBackButtonAnalyzer : DiagnosticAnalyzer
+    {
+        public const string DiagnosticId = "UA307";
+        private const string Category = "Fix";
+
+        private static readonly LocalizableString Title = "Custom back button implementation is needed";
+        private static readonly LocalizableString MessageFormat = "Back button '{0}' should be replaced with WinUI back button";
+        private static readonly LocalizableString Description = "Detect UWP back button";
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
+
+        private static readonly DiagnosticDescriptor Rule = new(DiagnosticId, Title, MessageFormat, Category, DiagnosticSeverity.Warning, isEnabledByDefault: true, description: Description);
+
+        internal const string FixStateProperty = "FixState";
+        internal const string FixStatePossible = "FixPossible";
+        internal const string FixStateNotPossible = "FixNotPossible";
+        internal const string FixStateComplete = "FixComplete";
+
+        public override void Initialize(AnalysisContext context)
+        {
+            if (context is null)
+            {
+                throw new ArgumentNullException(nameof(context));
+            }
+
+            context.EnableConcurrentExecution();
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.Analyze);
+
+            context.RegisterSyntaxNodeAction(AnalyzeLocalDeclarationStatement, SyntaxKind.SimpleAssignmentExpression);
+            context.RegisterSyntaxNodeAction(AnalyzeLocalDeclarationStatement, SyntaxKind.AddAssignmentExpression);
+        }
+
+        private void AnalyzeLocalDeclarationStatement(SyntaxNodeAnalysisContext context)
+        {
+            var node = (AssignmentExpressionSyntax)context.Node;
```

Comment:
may be `if (context.Node is AssignmentExpressionSyntax)` ?

Reply:
context.Node will always be of AssignmentExpressionSyntax type because that's the type we registered in the method right above. So the check is redundant